### ISSUE TITLE
Stream support?

### ISSUE THREAD

**Issue by christianbundy** [ungoldman/hyperaxe#11]
Hi! Me again. I'm rendering a webpage like:

```js
div(
  someArray
)
```

Unfortunately my `someArray` takes a long time to generate, so I'm considering replacing it with a stream. Is that sort of thing within scope of this project? To be explicit, I'd hope that I could pass `someStream` and have HyperAxe send `<div>` immediately followed by each stream item as soon as it comes through -- finally sending `</div>` when it's done. What do you think?

**Comment by ungoldman** [comment]
Depends on how much it would grow the current size of the package and impact the current API. If it's negligible then I don't see why not. How would you imagine implementing this?

**Comment by christianbundy** [comment]
I'm not sure -- I was hoping to hear your opinions on whether it's feasible or whether I'd need to patch something deeper (probably hyperscript). I guess my question is whether we *can* just stream the opening tags without the inner HTML or closing tags. I don't see a simple way to do that with hyperscript's API, but I figured I should ask you since you're more familiar.

**Comment by ungoldman** [comment]
Can't say it's something I've tried before -- I've used streams to pipe html to a client from a server context (using https://github.com/substack/hyperstream) but not in a browser context.

Hyperaxe is very much only designed for synchronous use to produce simple collections of DOM nodes for component-based architectures (https://github.com/choojs/choo being the primary use case for me). This usually means the state of the thing to be produced is already available, and hyperaxe is just the factory for producing DOM. If an array gets mutated or replaced with something larger I usually just rerender with the new data, possibly using some update strategies like DOM diffing (https://github.com/choojs/nanomorph or something like it).

I can't think of any examples of using a stream implementation in the browser to produce DOM content right now but it is very late here. So I don't really know off-hand how I'd approach it. If I think of something in the next few days I'll leave a comment here but otherwise I'm afraid my advice is as good as a random search at the moment. Best of luck 😅
